### ISSUE TITLE
Crash when deleting word during completion

### ISSUE THREAD

**Issue by jntrnr** [nushell/nushell#4436]
### Describe the bug

type `do { str<tab>` then do alt+backspace to delete the word when the completion menu is up

```
do { strthread 'main' panicked at 'removal index (is 2) should be < len (is 2)', crates/nu-cli/src/completions.rs:178:36
stack backtrace:
   0: rust_begin_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:107:14
   2: alloc::vec::Vec<T,A>::remove::assert_failed
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/alloc/src/vec/mod.rs:1397:13
   3: alloc::vec::Vec<T,A>::remove
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/alloc/src/vec/mod.rs:1402:13
   4: nu_cli::completions::NuCompleter::completion_helper
             at ./crates/nu-cli/src/completions.rs:178:29
   5: <nu_cli::completions::NuCompleter as reedline::completion::base::Completer>::complete
             at ./crates/nu-cli/src/completions.rs:367:26
   6: <reedline::menu::completion_menu::CompletionMenu as reedline::menu::Menu>::update_values
   7: reedline::engine::Reedline::handle_editor_event
             at /home/jt/.cargo/git/checkouts/reedline-e42026a78d91c510/8c565e4/src/engine.rs:725:25
   8: reedline::engine::Reedline::handle_event
src/engine.rs:454:13
   9: reedline::engine::Reedline::read_line_helper
  10: reedline::engine::Reedline::read_line
             at /home/jt/.cargo/git/checkouts/reedline-e42026a78d91c510/8c565e4/src/engine.rs:333:22
  11: nu::repl::evaluate
             at ./src/repl.rs:233:21
```

### How to reproduce

1. See above

### Expected behavior

Delete the word and do not crash

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.59.0                                   |
| branch             | known_external                           |
| short_commit       | 7c33d795                                 |
| commit_hash        | 7c33d79539df2d929474a34a06470d4dc87ba680 |
| commit_date        | 2022-02-11 14:46:46 +00:00               |
| build_os           | linux-x86_64                             |
| rust_version       | rustc 1.58.0 (02072b482 2022-01-11)      |
| rust_channel       | stable-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.58.0 (7f08ace4f 2021-11-24)      |
| pkg_version        | 0.59.0                                   |
| build_time         | 2022-02-11 09:47:51 -05:00               |
| build_rust_channel | debug                                    |
| features           | default, which                           |
| installed_plugins  | gstat, inc                               |

**Comment by jntrnr** [comment]
seems related to #4433. may not be present in main

**Comment by fdncred** [comment]
On my Mac, I tried to repo this but I don't get any completions when doing `do { str<tab>`. it just says `NO RECORDS FOUND`. So, I just did `do { str<tab><alt-backspace>` to see if that triggered a crash. It deleted `str`.

**Comment by jntrnr** [comment]
Great. So this was specific to the #4433 PR and didn't show up in main. I've gone ahead and put fixes for it into #4433 so we can land it and not have to chase this one down.
